tree/dataframe/test/dataframe_regression.cxx

```diff
@@ -56,6 +57,11 @@ TEST(RDFRegressionTests, AliasAndSubBranches)
    t.Fill();
    t.Fill();
 
+#ifndef NDEBUG
+   ROOT::TestSupport::CheckDiagsRAII diagRAII{kWarning, "RTreeColumnReader::Get",
+                                              "Branch topbranch.fCoordinates.fX hangs from a non-split branch. A copy "
+                                              "is being performed in order to properly read the content."};
+#endif
```

Comment:
Note that the reason for this warning has always been there, i.e. RDF has always needed to copy the branch in this particular test. The reason it was not visible before is that this warning only appears in builds that *do not* define the `NDEBUG` macro. Before this changes, RDF was always jitting the creation of the `RTreeColumnReader<T>` due to the filter operation. cling by default does not define `NDEBUG`, so that when the code that copies the branch value was reached, it did not trigger the warning. With this PR, the creation of the TTree column reader is not JITted anymore, so the warning is issued if the build of ROOT had the NDEBUG undefined (e.g. in `Debug` cmake builds).